lib/gulp-image-converter.js

```diff
@@ -0,0 +1,62 @@
+var path = require('path');
+var _ = require('underscore');
+var through = require('through2');
+var lwip = require('lwip');
+var File = require('vinyl');
+var gutil = require('gulp-util');
+
+function convert(options, file, cb) {
+  var format = path.extname(file.path).slice(1);
+  var targetFormat = options.format || format;
+
+  lwip.open(file.contents, format, function(err, image) {
+    if (err) {
+      return cb(err);
+    }
+
+    image.toBuffer(
+      targetFormat,
+      options.formatParams || {},
+      function(err, buffer) {
+        if (err) {
+          return cb(err);
+        }
+        var newFile = new File({
+          cwd: file.cwd,
+          base: file.base,
+          path: file.path.replace(format, targetFormat),
```

Comment:
Technically this will cause weird things to happen if the file path contains e.g. `.jpg` somewhere in the middle of it, rather than at the very end, but that seems quite unlikely.